src/main/java/net/mcreator/ui/modgui/VillagerProfessionGUI.java

```diff
@@ -0,0 +1,206 @@
+			TextureImportDialogs.importMultipleTextures(mcreator, TextureType.ENTITY);
+			professionTextureFile.removeAllItems();
+			professionTextureFile.addItem("");
+			mcreator.getFolderManager().getTexturesList(TextureType.ENTITY)
+					.forEach(el -> professionTextureFile.addItem(el.getName()));
+		});
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/profession_texture"),
+				L10N.label("elementgui.villager_profession.profession_texture")));
+		subpanel.add(PanelUtils.centerAndEastElement(professionTextureFile, importProfessionTexture));
+
+		JButton importZombifiedProfessionTexture = new JButton(UIRES.get("18px.add"));
+		importZombifiedProfessionTexture.setToolTipText(
+				L10N.t("elementgui.villager_profession.import_zombified_profession_texture"));
+		importZombifiedProfessionTexture.setOpaque(false);
+		importZombifiedProfessionTexture.addActionListener(e -> {
+			TextureImportDialogs.importMultipleTextures(mcreator, TextureType.ENTITY);
+			zombifiedProfessionTextureFile.removeAllItems();
+			zombifiedProfessionTextureFile.addItem("");
+			mcreator.getFolderManager().getTexturesList(TextureType.ENTITY)
+					.forEach(el -> zombifiedProfessionTextureFile.addItem(el.getName()));
+		});
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/zombified_profession_texture"),
+				L10N.label("elementgui.villager_profession.zombified_profession_texture")));
+		subpanel.add(PanelUtils.centerAndEastElement(zombifiedProfessionTextureFile, importZombifiedProfessionTexture));
+
+		page1group.addValidationElement(displayName);
+		page1group.addValidationElement(pointOfInterest);
+		page1group.addValidationElement(actionSound.getVTextField());
+		page1group.addValidationElement(professionTextureFile);
+
+		displayName.setValidator(new TextFieldValidator(displayName,
+				L10N.t("elementgui.villager_profession.profession_needs_display_name")));
+		displayName.enableRealtimeValidation();
+		pointOfInterest.setValidator(new PointOfInterestValidator(mcreator, pointOfInterest, savedPointOfInterest));
+		actionSound.getVTextField().setValidator(new TextFieldValidator(actionSound.getVTextField(),
+				L10N.t("elementgui.common.error_sound_empty_null")));
+		professionTextureFile.setValidator(() -> {
+			if (professionTextureFile.getSelectedItem() == null || professionTextureFile.getSelectedItem().equals(""))
+				return new Validator.ValidationResult(Validator.ValidationResultType.ERROR,
+						L10N.t("elementgui.villager_profession.profession_needs_texture"));
+			return Validator.ValidationResult.PASSED;
+		});
+
+		JPanel mainPanel = new JPanel(new BorderLayout(0, 50));
+		mainPanel.add("Center", subpanel);
+		mainPanel.setOpaque(false);
+
+		addPage(L10N.t("elementgui.common.page_properties"), PanelUtils.totalCenterInPanel(mainPanel));
+
+		if (!isEditingMode()) {
+			displayName.setText(StringUtils.uppercaseFirstLetter(modElement.getName().toLowerCase(Locale.ROOT)));
```

Comment:
This would be better with `StringUtils.machineToReadableName`, similar to the name field of blocks, items etc.